docs/semgrepignore-v2-reference.md

```diff
@@ -0,0 +1,155 @@
+---
+slug: semgrepignore-v2-reference
+append_help_link: true
+description: |
+  Reference of the semgrepignore file fitering mechanism for Semgrep
+hide_title: true
+title: Semgrepignore reference (v2)
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Semgrepignore reference (v2)
+
+This document covers the Semgrepignore target filtering system that is
+currently available with the `--experimental` option of the `semgrep`
+command ("v2"). It differs slightly from the legacy implementation which is
+referred to as "v1".
+
+## The target filtering process
+
+A `semgrep scan` command takes one or more scanning roots as
+arguments. The default scanning root is the current folder, `.`.
+Scanning roots are folders, individual files, or named pipes that should be
+expanded into a list of regular files to be analyzed. Symbolic links are
+allowed as scanning roots.
+
+Expanding a folder consists in listing its contents recursively with
+the following exceptions:
+
+* Symbolic links other than the original scanning roots are ignored.
+* In Git projects, Git submodules are ignored.
+* Paths excluded via Semgrepignore patterns are ignored. Semgrepignore
+  patterns can be of different sources which are detailed in the
+  upcoming section.
+
+The list of files obtained by expanding the scanning roots are called
+"target files". To obtain target files, Semgrep follows a
+number of fixed rules and some configurable filters.
+
+For each scanning root, Semgrep infers a project root (v2 only). The
+project root determines the location of applicable `.semgrepignore`
+files as well as `.gitignore` files in Git projects. In v1 where is no
+notion of a project root, the `.semgrepignore` file is unique and
+looked up in the current folder.
+
+<!-- TODO: explain project detection.
+     Go over options to disable listing files using `git ls-files`
+     while possibly still consulting the `.gitignore` files -- when we
+     have an option for it. Right now we have only `--no-git-ignore`
+     which is confusing and too coarse. I'd like to deprecate it as
+     soon as we have finer-grained replacements.
+-->
+
+As an experimental debugging aid, we provide the the `--x-ls` option
+to list the target files. `--x-ls-long` additionally prints excluded
+files and a brief justification. Beware that these two options are
```

Comment:
I suggest using the `caution` callout here for the "Beware..." sentence. 

Reference: https://docusaurus.io/docs/next/markdown-features/admonitions